app.rb

```diff
@@ -0,0 +1,158 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  MENU_OPTIONS = {
+    '1' => :list_books,
+    '2' => :list_people,
+    '3' => :create_person,
+    '4' => :create_book,
+    '5' => :create_rental,
+    '6' => :list_all_rentals,
+    '7' => :exit_program
+  }.freeze
+
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def start
+    loop { choose_option }
+  end
+
+  def list_books
+    if @books.empty?
+      puts 'No books available. Please add some books and try again.'
+    else
+      puts "There are #{@books.count} book(s) available."
+      @books.each_with_index do |book, index|
+        puts "#{index + 1}) Title: \"#{book.title}\" | Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_people
+    if @people.empty?
+      puts 'No people added yet. Please add a person and try again.'
+    else
+      @people.each_with_index do |person, index|
+        puts "#{index + 1}) [#{person.class}] Name: #{person.name} | Age: #{person.age} | ID: #{person.id}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Would you like to create a student (1) or a teacher (2)? Select a number: '
+    choice = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    case choice
+    when '1'
+      create_student(name, age)
+    when '2'
+      create_teacher(name, age)
+    else
+      puts 'Invalid choice. Please try again.'
+    end
+  end
+
+  def create_student(name, age)
+    print 'Grade: '
+    grade = gets.chomp
+    print 'Parent permission? [Y/N]: '
+    permission_option = gets.chomp.downcase
+    student = Student.new(grade, name, age, parent_permission: permission_option == 'y')
+    @people << student
+    puts "Student created successfully. ID is #{student.id}"
+  end
+
+  def create_teacher(name, age)
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(specialization, name, age)
+    @people << teacher
+    puts "Teacher created successfully. Teacher ID is #{teacher.id}"
+  end
```

Comment:
- [ ] **Good job so far 👏 👏**  However, it would be better to fix the error in creating a `teacher` to make your school library console app behave in the same way as the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) 🙏 

it brings the following error 👇 👇 
<img width="378" alt="Screen Shot 2023-10-19 at 20 06 14" src="https://github.com/axcell-0/school-library/assets/102757126/7318e4f5-0a24-4158-a7aa-9161dcd9a0c9">